### PR TITLE
Opg config cached

### PR DESCRIPTION
**Pull request by azime**
Jira; [NAV-2536](https://navitia.atlassian.net/browse/NAV-2536)

[NAV-2536]: https://navitia.atlassian.net/browse/NAV-2536?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: source/jormungandr/jormungandr/olympic_site_params_manager.py [hunk 1]
```diff
@@ -30,19 +30,43 @@
...
 AttractivityVirtualFallback = namedtuple("AttractivityVirtualFallback", "attractivity, virtual_duration")
 
 
+class ResourceS3Object:
+    def __init__(self, s3_object, instance_name):
+        self.s3_object = s3_object
+        self.instance_name = instance_name
+
+    def __repr__(self):
+        return "{}-{}-{}".format(
```
Comment (xlqian) [inline source/jormungandr/jormungandr/olympic_site_params_manager.py]: you can use the `ETag` as the unique key of you object

https://docs.aws.amazon.com/AmazonS3/latest/API/API_Object.html

File: source/jormungandr/jormungandr/olympic_site_params_manager.py [hunk 2]
```diff
@@ -30,19 +30,43 @@
...
+
 class OlympicSiteParamsManager:
-    def __init__(self, instance):
+    def __init__(self, instance, config):
         self.olympic_site_params = dict()
+        self.map_filename_last_modified = dict()
         self.instance = instance
+        self.bucket_name = config.get("name")
+        self.folder = config.get("folder", "olympic_site_params")
+        self.args = config.get("args", {"connect_timeout": 2, "read_timeout": 2, "retries": {'max_attempts': 0}})
```
Comment (xlqian) [inline source/jormungandr/jormungandr/olympic_site_params_manager.py:65]: can you render these args configurable?

File: source/jormungandr/jormungandr/olympic_site_params_manager.py [hunk 3]
```diff
@@ -111,32 +135,45 @@ def get_json_content(self, s3_object):
...
-                    self.str_datetime_time_stamp(json_content)
-                    self.olympic_site_params.update(json_content)
+                    resource_s3_object = ResourceS3Object(obj, instance_name)
+                    json_content = self.load_data(resource_s3_object)
+                    result.update(json_content)
         except Exception:
             logger.exception("Error on OlympicSiteParamsManager")
+        return result
+
+    def get_opg_params(self):
```
Comment (xlqian) [inline source/jormungandr/jormungandr/olympic_site_params_manager.py]: minor: to be more Pythonic, you can define this as a `property`

File: source/jormungandr/jormungandr/olympic_site_params_manager.py [hunk 4]
```diff
@@ -111,32 +135,45 @@ def get_json_content(self, s3_object):
...
+        self.fill_olympic_site_params_from_s3()
+        return self.olympic_site_params
+
+    def fill_olympic_site_params_from_s3(self):
+        if not self.instance.olympics_forbidden_uris:
+            return
+        logger = logging.getLogger(__name__)
+        if not self.bucket_name:
```
Comment (xlqian) [inline source/jormungandr/jormungandr/olympic_site_params_manager.py:184]: should be checked during construction...

File: source/jormungandr/jormungandr/olympic_site_params_manager.py (hunk 1 above)
Comment (azime) [reply source/jormungandr/jormungandr/olympic_site_params_manager.py]: Done

File: source/jormungandr/jormungandr/olympic_site_params_manager.py (hunk 2 above)
Comment (azime) [reply source/jormungandr/jormungandr/olympic_site_params_manager.py:65]: it's the cas

File: source/jormungandr/jormungandr/olympic_site_params_manager.py (hunk 3 above)
Comment (azime) [reply source/jormungandr/jormungandr/olympic_site_params_manager.py]: Done

**Review by pbench — APPROVED** [review]

**Pull request merged**